### PR TITLE
feat(crypto): Add Key Evolving Signatures (KES)

### PR DESCRIPTION
**Pull request by AndrewWestberg**
This code modifies block headers so that the HeaderBody bytes are wrapped in KeepRaw. These raw bytes are used in the validation of KES signatures.

pallas-crypto has been updated with wrappers around kes signatures for both evolving the secret key as well as verifying signatures with the public key.

### CODE REVIEW

File: pallas-crypto/src/kes/mod.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+/// KES error
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("KES error: {0}")]
+    Kes(#[from] kes_summed_ed25519::errors::Error),
+}
+
+/// KES secret key
+pub struct KesSecretKey<'a> {
```
Comment (scarmuega) [inline pallas-crypto/src/kes/mod.rs:14]: Is there any scrubbing mechanism in the upstream repo that we can use on Drop to ensure we clear private key data from memory?

File: pallas-crypto/src/kes/mod.rs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        // TODO: extend() could potentially re-allocate memory to a new location and copy the sk_bytes.
+        // This would leave the original memory containing the secret key without being wiped.
+        sk_bytes.extend([0u8; 4]); // default to period = 0
+        let sum_6_kes = Sum6Kes::from_bytes(sk_bytes.as_mut_slice())?;
+        Ok(KesSecretKey { sum_6_kes })
+    }
+
+    /// Get the internal representation of the KES secret key at the current period
+    /// This value will include the period as the last 4 bytes in big-endian format
+    pub fn as_bytes(&self) -> &[u8] {
```
Comment (scarmuega) [inline pallas-crypto/src/kes/mod.rs:30]: In other parts of the crypto crate we use the name `leak_into_bytes` and add a Safety disclaimer to make sure devs understand the risks of exposing these bytes. Maybe we could do the same here.

File: pallas-crypto/src/kes/mod.rs (hunk 1 above)
Comment (AndrewWestberg) [reply pallas-crypto/src/kes/mod.rs:14]: Yes. I added this here: https://github.com/txpipe/kes/pull/2

File: pallas-crypto/src/kes/mod.rs (hunk 2 above)
Comment (AndrewWestberg) [reply pallas-crypto/src/kes/mod.rs]: fixed.

**Review by scarmuega — APPROVED** [review]

**Pull request merged**